src/Marten/Linq/ContainmentWhereFragment.cs

```diff
@@ -98,5 +101,18 @@ public static IWhereFragment SimpleArrayContains(MemberInfo[] members, ISerializ
 
             throw new NotSupportedException();
         }
+
+        private static object GetMemberValue(MemberInfo member, object expressionValue, EnumStorage enumStorage)
+        {
+            var value = expressionValue;
+
+            if (member.MemberType is MemberTypes.Property && ((PropertyInfo)member).PropertyType.IsEnum
```

Comment:
@mysticmind There should be an extension method in the codebase called `GetMemberType()` that will deal w/ the PropertyInfo vs. FieldInfo differences here. You'll want to use that so that this fix applies just as well to public Fields

Reply:
@jeremydmiller sure, will update the logic using `GetMemberType()`